example/custom_configuration.cpp/custom_configuration.cpp

```diff
@@ -0,0 +1,184 @@
+        } else {
+            usage(argv[0]);
+            return 1;
+        }
+
+        connection_url = argv[1];
+        connection_result = dc.add_any_connection(connection_url);
+    } else if (argc == 4) {
+        uint8_t system_id = std::stoi(argv[2]);
+        uint8_t component_id = std::stoi(argv[3]);
+        Mavsdk::Configuration configuration(system_id, component_id);
+        dc.set_configuration(configuration);
+
+        connection_url = argv[1];
+        connection_result = dc.add_any_connection(connection_url);
+    } else {
+        usage(argv[0]);
+        return 1;
+    }
+
+    if (connection_result != ConnectionResult::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT
+                  << "Connection failed: " << connection_result_str(connection_result)
+                  << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+    // We don't need to specify the UUID if it's only one system anyway.
+    // If there were multiple, we could specify it with:
+    // dc.system(uint64_t uuid);
+    System& system = dc.system();
+
+    std::cout << "Waiting to discover system..." << std::endl;
+    dc.register_on_discover([&discovered_system](uint64_t uuid) {
+        std::cout << "Discovered system with UUID: " << uuid << std::endl;
+        discovered_system = true;
+    });
+
+    // We usually receive heartbeats at 1Hz, therefore we should find a system after around 2
+    // seconds.
+    sleep_for(seconds(2));
+
+    if (!discovered_system) {
+        std::cout << ERROR_CONSOLE_TEXT << "No system found, exiting." << NORMAL_CONSOLE_TEXT
+                  << std::endl;
+        return 1;
+    }
+
+    // Register a callback so we get told when components (camera, gimbal) etc
+    // are found.
+    system.register_component_discovered_callback(component_discovered);
+
+    auto telemetry = std::make_shared<Telemetry>(system);
+    auto action = std::make_shared<Action>(system);
+
+    // We want to listen to the altitude of the drone at 1 Hz.
+    const Telemetry::Result set_rate_result = telemetry->set_rate_position(1.0);
+    if (set_rate_result != Telemetry::Result::SUCCESS) {
+        std::cout << ERROR_CONSOLE_TEXT
+                  << "Setting rate failed:" << Telemetry::result_str(set_rate_result)
+                  << NORMAL_CONSOLE_TEXT << std::endl;
+        return 1;
+    }
+
+    // Set up callback to monitor altitude while the vehicle is in flight
+    telemetry->position_async([](Telemetry::Position position) {
+        std::cout << TELEMETRY_CONSOLE_TEXT // set to blue
+                  << "Altitude: " << position.relative_altitude_m << " m"
+                  << NORMAL_CONSOLE_TEXT // set to default color again
+                  << std::endl;
+    });
+
+    // Check if vehicle is ready to arm
+    while (telemetry->health_all_ok() != true) {
+        std::cout << "Vehicle is getting ready to arm" << std::endl;
+        sleep_for(seconds(1));
+    }
+
+    // Arm vehicle
```

Comment:
I don't think the rest of this example makes sense because if your configuration is `autopilot` you can't actually send arm, or at least you should not.

Also, it just confuses a reader because it doesn't really matter. Therefore, I would keep it short and remove all of the stuff down here. If it only does configuration and then nothing else, that's fine.